pycaption/srt.py

```diff
@@ -97,6 +97,33 @@ def write(self, caption_set):
         return caption_content
 
     def _recreate_lang(self, captions):
+        # Merge caption's that are on the exact same timestamp otherwise some
+        # players will play them in reversed order, libass specifically which is
+        # used quite a lot, including VLC and MPV.
+        new_captions = []
```

Comment:
I think we can refactor this to a simpler form, here's a suggestion:
```suggestion
merged_captions = [captions[0]] if captions else []
for caption in captions[1:]:
    # Merge if the timestamp is the same as last caption
    if (caption.start, caption.end) == (merged_captions[-1].start, merged_captions[-1].end):
    merged_captions[-1] = Caption(
        start=caption.start, end=caption.end,
        nodes=merged_captions[-1].nodes + caption.nodes)
   else:
       # Different timestamp, end of merging, append new caption
       merged_captions.append(caption)
captions = merged_captions
```

Reply:
This approach assumes the identical timestamps will be next to each other in the file, I'm not sure if we're supposed to treat the case where same timestamp appears in random order, that sounds more like a caption issue, rather than a conversion issue.

Reply:
Done! Please have a look.